File: app/src/main/java/com/softaai/designpatterns/java/structural/adapter/Mp4Player.java
package com.softaai.designpatterns.java.structural.adapter;

/**
 * Created by Amol Pawar on 02-04-2019.
 * softAai Apps
 */
public class Mp4Player implements AdvancedMediaPlayer {
    @Override
    public void playVlc(String fileName) {

    }

    @Override
    public void playMp4(String fileName) {
       System.out.println("Playing mp4 file. Name : " + fileName);
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/structural/adapter/Mp4Player.kt
package com.softaai.designpatterns.kotlin.structural.adapter


/**
 * Created by Amol Pawar on 02-04-2019.
 * softAai Apps
 */
class Mp4Player : AdvancedMediaPlayer {
    override fun playVlc(fileName: String) {
    }

    override fun playMp4(fileName: String) {
       println("Playing mp4 file. Name : " + fileName)
    }
}